src/main/java/seedu/duke/parsers/GradeParser.java

```diff
@@ -18,14 +18,15 @@
     // Unescaped regex for testing:
     // ((?<moduleCode>\w+)(\s+(?<moduleGrade>(?i)(CU|CS|[A-B][+-]?|[C-D][+]?|F|S|U))))(?<invalid>.*)
     private static final String GRADE_FORMAT = "((?<moduleCode>\\w+)(\\s+"
-            + "(?<moduleGrade>(?i)(CU|CS|[A-B][+-]?|[C-D][+]?|F|S|U))))(?<invalid>.*)";
+            + "(?<moduleGrade>(?i)(CU|CS|[A-B][+-]?|[C-D][+]?|F|S|U)|(?<invalidModuleGrade>.*))))(?<invalid>.*)";
```

Comment:
The regex should make invalid always be null, maybe invalid can be removed in this case

Reply:
Nop, in this case, the invalid will be capturing excess arguments at the back of each command.

For eg,

![Screenshot (867)](https://user-images.githubusercontent.com/69583592/161413045-4871bb0e-f8f6-4b07-9e7e-352c5defb8be.png)